modern/src/RegisterDialog.js

```diff
@@ -0,0 +1,109 @@
+import t from './common/localization'
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import TextField from '@material-ui/core/TextField';
+import Alert from '@material-ui/lab/Alert';
+
+const RegisterDialog = ({ open, onResult }) => {
+  const [formFields, setFormFields] = useState({});
```

Comment:
Why not have separate fields?

Reply:
I didn't change, I think this way code is more compact.

Reply:
I think we should separate fields to match what we have in other places. And errors is not needed at all, I think.

Reply:
I will separate fields.